adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -31,10 +31,11 @@ @implementation GADMAdapterUnityRewardedAd {
   GADMediationAdConfiguration *_adConfiguration;
 
   /// An ad event delegate to invoke when ad rendering events occur.
-  id<GADMediationRewardedAdEventDelegate> _adEventDelegate;
+  __weak id<GADMediationRewardedAdEventDelegate> _adEventDelegate;
```

Comment:
While delegate's are typically weak, we typically make this one strong in other adapters ([example](https://github.com/googleads/googleads-mobile-ios-mediation/blob/de680e4d4485d50949e5914bf6eefe91d2dbedff/adapters/Facebook/FacebookAdapter/OpenBidding/GADFBRewardedRenderer.m#L36-L39)). Unlike most delegate patterns where you pass a delegate to some other class, who should hold a weak reference to it, in this case Google's SDK is *returning* a delegate object to you from a completion handler. __weak only works because Google's SDK also happens to have a strong reference to the object it returned, but as a best practice we keep this explicitly strong just in case that ever changes.

Hopefully that doesn't affect any memory leak analysis you were doing, but please let me know if you noticed otherwise.

Reply:
We added `__weak` because without it we were seeing objects hanging around.  But I will take a look at the example you provided and double check what is going on without `__weak` just to be sure it is needed.

Reply:
@ericleich I just confirmed that if we don't do this then we have a retain cycle.

<img width="1904" alt="Screen Shot 2021-04-16 at 11 05 58 AM" src="https://user-images.githubusercontent.com/54249463/115070545-af794c80-9eba-11eb-9079-c914f0844314.png">

We attempted to modify our code to be more similar to the example provided, but we still experienced the same retain cycle.

We then attempted to add a dispatch after to test breaking the retain cycle manually but it resulted a different retain cycle.

<img width="1519" alt="Screen Shot 2021-04-16 at 1 46 16 PM" src="https://user-images.githubusercontent.com/54249463/115070561-b6a05a80-9eba-11eb-91b7-a1fac7594af7.png">

```
- (void)loadRewardedAdForAdConfiguration:(GADMediationRewardedAdConfiguration *)adConfiguration
                       completionHandler:
                           (GADMediationRewardedLoadCompletionHandler)completionHandler {
  self.rewardedAd = [[GADMAdapterUnityRewardedAd alloc] init];
  [self.rewardedAd loadRewaredeAdWithAdConfiguration:adConfiguration completionHandler:completionHandler];
  
  NSInteger ttl = 10;
  dispatch_after(dispatch_time(DISPATCH_TIME_NOW, ttl * NSEC_PER_SEC), dispatch_get_main_queue(), ^{
    self.rewardedAd = nil;
  });
}
```

Do you have any other suggestions to try, or should we leave it `__weak` for now?

Reply:
@richardh-unity3d Thanks for the detailed writeup. `__weak` sounds good for now to unblock this PR, and `__weak` in general is still expected to work. I'll also sync with the team internally on this a bit more, as my previous understanding is strong would not cause this.

One interesting thing is I can't find even find GADMRewardedAdNetworkConnector in our SDK head code -- which Google SDK version are you using in your test?

Reply:
Thanks @ericleich.  It looks like my Podfile has `pod 'Google-Mobile-Ads-SDK', '~> 7.69.0'` in it.

Reply:
@richardh-unity3d got it. I confirmed `GADMRewardedAdNetworkConnector` was removed in version 8.0.0, but I'm still trying to unpack why it was referenced even in your test against 7.69.0. That class was used for our legacy rewarded medation API which the adapter doesn't implement.

Could you try running against 8.0.0 or higher and seeing if you still see the retain cycle with the strong reference? If you need a test app updated for version 8.0.0, you can use https://github.com/googleads/googleads-mobile-ios-examples/tree/master/Objective-C/admob/RewardedVideoExample.

Reply:
confiirmed internally that pre v8.0.0 there was some complexity in routing rewarded mediation, but in v8.0.0, that retain cycle shouldn't be there.